src/trie.lisp

```diff
@@ -59,21 +59,30 @@
   (labels ((add-node (children-list char)
              ;; Insert node in lexicographical order
              (let ((new-node (make-instance 'trie :key char)))
-               (list new-node (sort (cons new-node children-list)
-                                    #'char-greaterp :key #'key)))))
+               (values new-node
+                       (if (null children-list)
```

Comment:
Please add a comment to explain this operation (or better yet, create a named local function for this purpose).